src/libmongoc/doc/includes/returns-cursor.txt

```diff
@@ -0,0 +1,8 @@
+This function returns a newly allocated :symbol:`mongoc_cursor_t` that should be freed with :symbol:`mongoc_cursor_destroy()` when no longer in use. The returned :symbol:`mongoc_cursor_t` is never ``NULL``; if the parameters are invalid, the :symbol:`bson:bson_error_t` in the :symbol:`mongoc_cursor_t` is filled out, and the :symbol:`mongoc_cursor_t` is returned before the server is selected. The user must call :symbol:`mongoc_cursor_next()` on the returned :symbol:`mongoc_cursor_t` to execute the aggregation pipeline.
```

Comment:
```suggestion
This function returns a newly allocated :symbol:`mongoc_cursor_t` that should be freed with :symbol:`mongoc_cursor_destroy()` when no longer in use. The returned :symbol:`mongoc_cursor_t` is never ``NULL``, even on error. The user must call :symbol:`mongoc_cursor_next()` on the returned :symbol:`mongoc_cursor_t` to execute the initial command.
```
I realize this came from existing documentation :) but I think it is best to avoid mentioning the `bson_error_t` inside the `mongoc_cursor_t`. `mongoc_cursor_t` is an opaque type, so it seems preferable to just note that an error could be set on the returned cursor (before the first call to `mongoc_cursor_next`.

Cursor returning functions like `mongoc_collection_find_with_opts` would run a find command initially instead of an aggregate command, so I think it is preferable to be generic.